streams/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java

```diff
@@ -291,48 +304,55 @@ public GroupAssignment assign(final Cluster metadata, final GroupSubscription gr
         }
     }
 
-    private static class FutureSubscriptionInfo extends SubscriptionInfo {
+    private static class FutureSubscriptionInfo {
```

Comment:
If we use this class only for its explicit purpose (encoding the SubscriptionInfo for the version probing test), and not couple it to the SubscriptionInfo, then we won't create a nightmare for ourselves when we want to actually change the SubscriptionInfo class (for example, in KIP-441).

As always, minimizing coupling improves maintainability.